test/test_matmul_cuda.py

```diff
@@ -175,7 +176,53 @@ def _expand_to_batch(t: torch.Tensor):
         self.assertEqual(out1_gpu, out2_gpu[0])
 
 
+@unittest.skipIf(TEST_WITH_ROCM, "FP8 is not supported on ROCM")
+@unittest.skipIf(not torch.cuda.is_available() or torch.cuda.get_device_capability() < (9, 0), "FP8 is only supported on H100+")
+class TestFP8MatmulCuda(TestCase):
+    def _test_tautological_mm(self, device: str = "cuda",
+                              x_dtype: torch.dtype = torch.float8_e4m3fn,
+                              y_dtype: torch.dtype = torch.float8_e4m3fn,
+                              out_dtype: Optional[torch.dtype] = None,
+                              size: int = 16) -> None:
+        x_fp8 = torch.rand(size, size, device=device).to(x_dtype)
+        y_fp8 = torch.eye(size, device=device).to(y_dtype).t()
+        out_fp32 = torch.mm(x_fp8.to(torch.float), y_fp8.to(torch.float))
+        (out_fp8, amax_fp8) = torch._scaled_mm(x_fp8, y_fp8, out_dtype=out_dtype)
+        if out_dtype is not None:
+            self.assertEqual(out_dtype, out_fp8.dtype)
+        if out_dtype not in [torch.float16, torch.bfloat16, torch.float]:
+            self.assertEqual(out_fp32.amax(), amax_fp8)
+        self.assertEqual(out_fp32, out_fp8.to(torch.float))
+
+    def test_float8_basics(self, device) -> None:
+        self._test_tautological_mm(device, torch.float8_e4m3fn, torch.float8_e4m3fn, size=16)
+        self._test_tautological_mm(device, torch.float8_e4m3fn, torch.float8_e5m2, size=32)
+        self._test_tautological_mm(device, torch.float8_e5m2, torch.float8_e4m3fn, size=48)
+        # According to https://docs.nvidia.com/cuda/cublas/#id99 8F_E5M2 MM is unsupported
+        with self.assertRaises(RuntimeError):
+            self._test_tautological_mm(device, torch.float8_e5m2, torch.float8_e5m2)
+
+    def test_float8_out_dtype(self, device) -> None:
+        self._test_tautological_mm(device, size=64, out_dtype=torch.float16)
+        self._test_tautological_mm(device, size=96, out_dtype=torch.float32)
+        self._test_tautological_mm(device, size=80, out_dtype=torch.bfloat16)
+        with self.assertRaises(RuntimeError):
+            self._test_tautological_mm(device, out_dtype=torch.float8_e5m2)
+
+    def test_float8_scale(self, device) -> None:
+        size = (16, 16)
```

Comment:
If I change this to (7x7) I get:
```
RuntimeError: CUDA error: CUBLAS_STATUS_NOT_SUPPORTED when calling `cublasLtMatmulAlgoGetHeuristic( ltHandle, computeDesc.descriptor(), Adesc.descriptor(), Bdesc.descriptor(), Cdesc.descriptor(), Ddesc.descriptor(), preference.descriptor(), 1, &heuristicResult, &returnedResult)`

```

Reply:
Yep, see limitations line: `Only works for matrix sizes divisible by 16`